interp/cfg.go

```diff
@@ -860,7 +883,12 @@ func (interp *Interpreter) cfg(root *node, importPath string) ([]*node, error) {
 					n.gen = nop
 					n.findex = -1
 					n.typ = c0.typ
-					n.rval = c1.rval.Convert(c0.typ.rtype)
+					if c, ok := c1.rval.Interface().(constant.Value); ok {
+						i, _ := constant.Int64Val(constant.ToInt(c))
+						n.rval = reflect.ValueOf(i).Convert(c0.typ.rtype)
+					} else {
+						n.rval = c1.rval.Convert(c0.typ.rtype)
+					}
```

Comment:
Does this have anything to do with the shift bug? If not, what is it about?

Reply:
this indeed relates to a different issue, but related also to shift, see interp_eval_test.go:78 for the corresponding test.

Reply:
ok. also, I'm not 100% sure, but I think instead of an if/else, we can have an if and break early here.

Reply:
not worth for such small blocks